server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -3542,9 +3541,7 @@ public Vlan createVlanAndPublicIpRange(final long zoneId, final long networkId,
                     continue;
                 }
                 // from here, subnet overlaps
-                if (!UriUtils.checkVlanUriOverlap(
-                        BroadcastDomainType.getValue(BroadcastDomainType.fromString(vlanId)),
-                        BroadcastDomainType.getValue(BroadcastDomainType.fromString(vlan.getVlanTag())))) {
+                if (!vlanId.equals(vlan.getVlanTag())) {
```

Comment:
I think this may cause an issue, because the vlan.getVlanTag() could be something like a range 100-200, so if vlanId is 101 the check should be done in the range. Let me revisit this shortly and check if the overlap check is necessary or not.

Reply:
Or perhaps if the issue is just for the 'untagged' VLAN, a specific check/condition could be added. This is because sometimes people add address ranges belonging to a VLAN (say vlan://101) instead of vlan://untagged (or untagged)